### DISCUSSION TITLE
Suggestions for using Cloudy with a Video View?

### DISCUSSION THREAD

**Discussion by kiranrao-somnox** [skydoves/Cloudy#10] [Q&A]
Hi, I have a video player (implemented using ExoPlayer). When the video is paused, we show an overlay with controls to resume/replay the video. 

My design calls for blurring the paused frame for showing this overlay. I attempted to do this using `Cloudy`. However no matter what I do I cannot get the blur to appear/disappear based on whether the video is paused or not.

## Initial attempt

My initial attempt was this:

```kt
val videoPlayerSection: @Composable () -> Unit = {
    AndroidView(
        factory = {
            // create ExoPlayer's PlayerView here
        }
    ) { playerView ->
        // Do some things based on state updates here
    }
}


if (isPaused) {
    Cloudy {
        videoPlayerSection()
    } else {
        videoPlayerSection()
    }
}
```

Essentially it extracts the videoPlayerSection into a composable lamda and then if the video is paused it renders that lamda inside Cloudy, else renders it directly.

However this does not work as expected; it results in the factory being called every time the isPaused status changes so in my case it starts playing the video from start every time (since my `factory` block resets the video to the beginning).

## Second attempt

For my second attempt, I abandoned trying to extract the Video playing into a composable lambda. Instead I tried to get Cloudy to blur or not blur using the radius

```kt
Cloudy(radius = if (paused) 10 else 1) {
    MyVideoPlayer()
}
```

This also doesn't seem to work. I see a white overlay on top of my video. I don't know if there exists a blur radius that means "no blur".

Does anyone have any suggestions? Is this even supposed to work with a VideoView at all?

### Footnote

Here is what my design calls for:

<img src="https://user-images.githubusercontent.com/96243005/224319663-b796e872-d34d-4319-85a0-30c8e187427f.png" width="25%" height="25%">

**Comment by oguzhanaslann** [comment]
I dont know if this works but

- retrieve frames from video ( you may benefit from my blog [here](https://medium.com/@oguzhanaslann/dynamically-retrieving-frames-in-android-a659ab406786) 
- show them one by one on an image view placed on top of video view
- apply blur on the retrieved frames.

you can just show or hide the image view here based on your use case.
